### PR TITLE
feat(bleed): Fixed anti-ice fault logic

### PR DESCRIPTION
**Pull request by Eagle941**
## Summary of Changes
Fixed logic for the ANTI-ICE FAULT light

## Screenshots (if necessary)

None

## References

None

## Additional context

Following the integration of the new WAI model, it has been noted that the FAULT light remains stuck when the wing anti-ice valve doesn't open and the WAI is eventually switched OFF. This was due to an if-statement that didn't cater for all the combinations.

Further to the above, the timer of the wing anti-ice test runs even if the system is unpressurised. This is against any logic, but this is how I interpret the schematics (30-11-01).

Discord username (if different from GitHub): pythonist#6593

## Testing instructions

Added unit test to cater for this case.

If you wish to verify, you can:
- Set the plane Cold and Dark
- Turn BATT ON
- Turn APU ON (No BLEED)
- Click WAI
- [ ]  Verify ON and FAULT light
- The FAULT light will not go away because pneumatics aren't pressurised.
- Wait for 30 sec
- Click WAI
- [ ]  Verify ON and FAULT lights disappear

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Eagle941** [comment]
This fix belongs to the new WAI model, should I commit an update to the CHANGELOG?

**Review by BlueberryKing — APPROVED** [review]
Nice fix!

**Comment by BlueberryKing** [comment]
> This fix belongs to the new WAI model, should I commit an update to the CHANGELOG?

I don't think there is a convention for this, but a fix does not warrant a changelog entry in my opinion.

File: src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs [hunk 1]
```diff
@@ -411,16 +411,11 @@ impl WingAntiIceSystem {
...
-        // system to be considered on without a fault.
-        if self.wai_valve_controller.controller_signals_on() {
-            // If a controller signals `on` while its corresponding valve is closed
-            // this means the system has a fault.
-            if self.is_wai_valve_closed() {
-                self.wai_has_fault = true;
-            } else {
-                self.wai_has_fault = false;
-            }
+        // If valve doesn't follow controller, there is a fault
```
Comment (Gurgel100) [inline src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs:414]: Just a small change:
```suggestion
        self.wai_has_fault = self.wai_valve_controller.controller_signals_on() == self.is_wai_valve_closed();
```

**Review by Gurgel100 — COMMENTED** [review]
Just a small suggestion

File: src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs (hunk 1 above)
Comment (Eagle941) [reply src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs:414]: True! Thank you for spotting it

**Review by Gurgel100 — APPROVED** [review]
LGTM

**Comment by tracernz** [comment]
> > This fix belongs to the new WAI model, should I commit an update to the CHANGELOG?
> 
> I don't think there is a convention for this, but a fix does not warrant a changelog entry in my opinion.

Changelog is relative to stable releases, so no entry unless you fixed something that was broken in the previous stable release (or added/improved a feature since the previous stable)... that's 0.8 at the moment.

**Comment by Eagle941** [comment]
> > > This fix belongs to the new WAI model, should I commit an update to the CHANGELOG?
> > 
> > 
> > I don't think there is a convention for this, but a fix does not warrant a changelog entry in my opinion.
> 
> Changelog is relative to stable releases, so no entry unless you fixed something that was broken in the previous stable release (or added/improved a feature since the previous stable)... that's 0.8 at the moment.

I see this fix has been assigned to v0.10.0, but the new WAI model is planned for v0.9.0. Can this fix be moved to v0.9.0?

**Comment by tracernz** [comment]
Not sure who's acting as release manager for 0.9, but this probably needs cherry-picked to 0.9 branch in that case.

**Comment by BlueberryKing** [comment]
Non Quality Assurance Tester/Trainee Report

Discord: BlueberryKing#6641
Object of testing: #7562
Tier of Testing: 1
Date: 27/10/2022

Testing Process: 
- Spawned in Cold&Dark
- Powered up the airplane (no bleed, no engines)
- Turned on WAI
- Observed FAULT light and ON light, WAI valves closed on BLEED page
- Observed FAULT light turning off after 30s, ON light staying on
- Turned off WAI
- Observed ON light disappearing and no FAULT light

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7562
Tier of Testing  : 1
Date             : 28/10/2022

Testing Process: 
Set the plane Cold and Dark
Turn BATT ON
Turn APU ON (No BLEED)
Click WAI
 Verify ON and FAULT light - **OK**
The FAULT light will not go away because pneumatics aren't pressurised.
Wait for 30 sec
Click WAI
 Verify ON and FAULT lights disappear - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by LeDerpLegend** [comment]
Quality Assurance Tester Report

Discord : LeDerpLegend#0109
Object of testing: https://github.com/flybywiresim/a32nx/pull/7562
Tier of Testing : 1
Date : 28/10/2022

Spawned in Cold and Dark
Turned BATT ON
Turned APU ON with no BLEED
Clicked WAI
FAULT light turned on and verified WAI was on, BLEED page matched up
_The FAULT light will not go away because pneumatics aren't pressurized_ (VERIFIED)
Waited for 30 sec to pass
Clicked WAI again 
WAI ON indicator and FAULT indicator lights doused.

Negatives:
N/A

Testing Results:
PASSED

Conclusions:
Works as intended

**Pull request merged**